### ISSUE TITLE
[display] too much stuff (2)

### ISSUE THREAD

**Issue by Gama11** [HaxeFoundation/haxe#8643]
Toplevel completion includes .hx files that are not part of the classpath, such as these unit test classes in a Flixel project (where flixel is installed from GitHub to actually include these files):

![](https://i.imgur.com/UW7c4vw.png)

This didn't happen with rc.3. Probably this means that `readClassPaths` now explores things it shouldn't / didn't used to?

**Comment by Simn** [comment]
Partially fixed by https://github.com/HaxeFoundation/haxe/commit/54a0466f598b4215ce742cec47816a7f8ecd21eb. Have to check the moduleCreated case still.

**Comment by Gama11** [comment]
I don't think that fix was related to this issue? And there's no moduleCreated here.

**Comment by Simn** [comment]
What exactly is the setup to reproduce this? I'm trying HaxeRepro with a `-lib flixel` in build.hxml and these files don't show up.

**Comment by Gama11** [comment]
Not sure if `-lib flixel` is enough or if you need a proper Flixel project built with Lime. Did you install Flixel from GitHub so it actually has the test classes?

**Comment by Simn** [comment]
```
C:\Users\simn>haxelib path flixel
C:\GitHub\flixel\
-D flixel=4.6.2
```

**Comment by Gama11** [comment]
Hm.. well, actually I can reproduce it with a simple `-lib flixel` in HaxeRepro. The selected target doesn't really seem to matter.

**Comment by kLabz** [comment]
I cannot reproduce this exact issue, however I know similar issues can happen with libraries that don't use `classPath` field of `haxelib.json`, when used as `dev` or `git` versions (usually releases are fine because they only include relevant files).

I'm not sure Haxe can do more than it's doing already, and I'm also not sure how to reproduce this issue (even with older versions) in order to add a test.
